cwltool/workflow.py

```diff
@@ -398,11 +400,22 @@ def __init__(
             else:
                 nesting = 1
 
-            for _ in range(0, nesting):
+            for n in range(0, nesting):
                 for oparam in outputparms:
+                    if conditional and n == 0:
+                        oparam["type"] = ["null"] + cast(
```

Comment:
Can you add a test case that triggers this line?

Reply:
Yes, I was going to add a test, I just wanted to see what the other code checks had to say about it.